### PR TITLE
#245: Tests for AddProposedSolution

### LINKED ISSUE #245: Backend - Test addProposedSolution (part 1)
### Description

Write unit tests for some of the cases of the addProposedSolution endpoint. To work on this ticket, you will need to read through the endpoint to learn how it works.

### Acceptance Criteria

The following cases should be covered:
* the submitter does not exist
* the associated change request does not exist

### Proposed Solution

Check out this epic https://github.com/Northeastern-Electric-Racing/FinishLine/issues/238 for resources and tips on how to write these tests.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by ibarra000**
## Changes

Wrote unit tests for some of the cases of the addProposedSolution endpoint

## Test Cases

-  CR not found
-  User Access Denied
-  CR deleted
-  CR already accepted
-  ScopeCR not found
-  Accepted proposed solution

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #245

### CODE REVIEW

File: src/backend/tests/change-requests.test.ts [hunk 1]
```diff
@@ -288,4 +288,68 @@ describe('Change Requests', () => {
       expect(prisma.user_Settings.findUnique).toHaveBeenCalledTimes(1);
     });
   });
+  describe('addProposedSolution', () => {
```
Comment (Peyton-McKee) [inline src/backend/tests/change-requests.test.ts:291]: linebreak

File: src/backend/tests/change-requests.test.ts [hunk 2]
```diff
@@ -288,4 +288,68 @@ describe('Change Requests', () => {
...
+      jest.spyOn(prisma.change_Request, 'findUnique').mockResolvedValue(prismaChangeRequest1);
+      jest.spyOn(prisma.scope_CR, 'findUnique').mockResolvedValue(null);
+      await expect(() =>
+        ChangeRequestsService.addProposedSolution(aquaman, crId, budgetImpact, description, timelineImpact, scopeImpact)
+      ).rejects.toThrow(new NotFoundException('Change Request', crId));
+      expect(prisma.change_Request.findUnique).toHaveBeenCalledTimes(1);
+      expect(prisma.scope_CR.findUnique).toHaveBeenCalledTimes(1);
+    });
+
+    test('accepted addProposedSolution', async () => {
```
Comment (Peyton-McKee) [inline src/backend/tests/change-requests.test.ts:344]: dont need it to be one word

File: src/backend/tests/change-requests.test.ts (hunk 1 above)
Comment (Peyton-McKee) [reply src/backend/tests/change-requests.test.ts:291]: Also dont need it to be one word

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Excellent work!! A couple clearing up things

**Comment by ibarra000** [comment]
Fixed the changes!

**Review by Peyton-McKee — APPROVED** [review]
WWWWWWWWWW

**Pull request merged**